### DISCUSSION TITLE
Enabling RTSPS?

### DISCUSSION THREAD

**Discussion by ciaranmd212** [bluenviron/mediamtx#3036] [Questions]
### Question

I have a MediaMTX instance running on my laptop. I have an Amcrest IP camera. I can publish the camera's RTSP stream to the server so that the stream can be accessed from the MediaMTX instance.

However, I want to encrypt the stream. I have followed the steps for how TLS encryption can be enabled for incoming and outgoing RTSP streams by generating a server.key and server.crt using the commands from the GitHub page and editing lines in the .yml configuration file.

I have set up the following path for my IP camera in the .yml:

```
###############################################
# Path settings

# Settings in "paths" are applied to specific paths, and the map key
# is the name of the path.
# Any setting in "pathDefaults" can be overridden here.
# It's possible to use regular expressions by using a tilde as prefix,
# for example "~^(test1|test2)$" will match both "test1" and "test2",
# for example "~^prefix" will match all paths that start with "prefix".
paths:

  cam1:
    source: rtsp://user:password@192.168.68.142:554
    runOnInit: ffmpeg -i rtsp://user:password@192.168.68.142:554 -c:v copy -c:a copy -f rtsp rtsps://localhost:8322/stream/cam1
```

But I am getting the following output from the console log when it runs:

`2024/02/17 18:44:14 INF MediaMTX v1.5.1
2024/02/17 18:44:14 INF configuration loaded from C:\Users\PC1\Desktop\RTSPServer\mediamtx.yml
2024/02/17 18:44:14 INF [path cam1] [RTSP source] started
2024/02/17 18:44:14 INF [path cam1] runOnInit command started
2024/02/17 18:44:14 INF [RTSPS] listener opened on :8322 (TCP)
2024/02/17 18:44:14 INF [RTMP] listener opened on :1935
2024/02/17 18:44:14 INF [HLS] listener opened on :8888
2024/02/17 18:44:14 INF [WebRTC] listener opened on :8889 (HTTP), :8189 (ICE/UDP)
2024/02/17 18:44:14 INF [SRT] listener opened on :8890 (UDP)
ffmpeg version 2024-01-24-git-00b288da73-full_build-www.gyan.dev Copyright (c) 2000-2024 the FFmpeg developers
  built with gcc 12.2.0 (Rev10, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --pkg-config=pkgconf --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-bzlib --enable-lzma --enable-libsnappy --enable-zlib --enable-librist --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-libbluray --enable-libcaca --enable-sdl2 --enable-libaribb24 --enable-libaribcaption --enable-libdav1d --enable-libdavs2 --enable-libuavs3d --enable-libzvbi --enable-librav1e --enable-libsvtav1 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxvid --enable-libaom --enable-libjxl --enable-libopenjpeg --enable-libvpx --enable-mediafoundation --enable-libass --enable-frei0r --enable-libfreetype --enable-libfribidi --enable-libharfbuzz --enable-liblensfun --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-dxva2 --enable-d3d11va --enable-libvpl --enable-libshaderc --enable-vulkan --enable-libplacebo --enable-opencl --enable-libcdio --enable-libgme --enable-libmodplug --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libshine --enable-libtheora --enable-libtwolame --enable-libvo-amrwbenc --enable-libcodec2 --enable-libilbc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-ladspa --enable-libbs2b --enable-libflite --enable-libmysofa --enable-librubberband --enable-libsoxr --enable-chromaprint
  libavutil      58. 36.101 / 58. 36.101
  libavcodec     60. 38.100 / 60. 38.100
  libavformat    60. 20.100 / 60. 20.100
  libavdevice    60.  4.100 / 60.  4.100
  libavfilter     9. 17.100 /  9. 17.100
  libswscale      7.  6.100 /  7.  6.100
  libswresample   4. 13.100 /  4. 13.100
  libpostproc    57.  4.100 / 57.  4.100
2024/02/17 18:44:14 INF [path cam1] [RTSP source] ready: 2 tracks (H264, MPEG-4 Audio)
Input #0, rtsp, from 'rtsp://admin:majugarzet@192.168.68.142:554':
  Metadata:
    title           : Media Server
  Duration: N/A, start: 0.000000, bitrate: N/A
  Stream #0:0: Video: h264 (Main), yuv420p(progressive), 1920x1080, 100 tbr, 90k tbn
  Stream #0:1: Audio: aac (LC), 16000 Hz, mono, fltp
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
  Stream #0:1 -> #0:1 (copy)
2024/02/17 18:44:15 INF [RTSPS] [conn [::1]:57789] opened
2024/02/17 18:44:15 INF [RTSPS] [session aca5538f] created by [::1]:57789
2024/02/17 18:44:15 INF [RTSPS] [conn [::1]:57789] closed: path 'stream/cam1' is not configured
2024/02/17 18:44:15 INF [RTSPS] [session aca5538f] destroyed: not in use
[out#0/rtsp @ 000001647c1eeec0] Could not write header (incorrect codec parameters ?): Server returned 400 Bad Request
Conversion failed!
2024/02/17 18:44:15 INF [path cam1] runOnInit command exited: command exited with code 3486501640`

I'm using version 1.5.1.

I have no errors and was able to publish the stream with the same path configuration (using rtsp://, not rtsps://) when using no encryption. What possible solutions can I try?

**Comment by aler9** [comment]
This question has nothing to do with enabling or disabling anything: you are publishing to path "/cam1" but you are trying to read path "/stream/cam1", which of course doesn't exist.

**Reply by ciaranmd212** [reply]
Sorry. I have fixed that mistake, but I am still receiving an error.

Here is my fixed path:
`paths:

    cam1:
        source: rtsp://admin:majugarzet@192.168.68.142:554
        runOnInit: ffmpeg -i rtsp://user:password@192.168.68.142:554 -c:v copy -c:a copy -f rtsp rtsps://localhost:8322/cam1`


And this is the new error I am getting:

`Stream mapping:
  Stream #0:0 -> #0:0 (copy)
  Stream #0:1 -> #0:1 (copy)
2024/02/19 19:49:23 INF [RTSPS] [conn [::1]:50268] opened
2024/02/19 19:49:23 INF [RTSPS] [session 264651ce] created by [::1]:50268
2024/02/19 19:49:23 INF [RTSPS] [conn [::1]:50268] closed: can't publish to path 'cam1' since 'source' is not 'publisher'
2024/02/19 19:49:23 INF [RTSPS] [session 264651ce] destroyed: not in use
[out#0/rtsp @ 000001d22da7a040] Could not write header (incorrect codec parameters ?): Server returned 400 Bad Request
Conversion failed!
2024/02/19 19:49:24 INF [path cam1] runOnInit command exited: command exited with code 3486501640`

It works with no encryption. Is there anything I need to do for this?

**Reply by aler9** [reply]
i sincerely don't know what you are trying to achieve, but the README contains instructions on how to ingest streams from any source and read them with any protocol. Conversion between protocols is automatic and the command inside `runOnInit` you used is not mentioned anywhere. Please take a look at the README in order to find out how to ingest streams from a RTSP camera and read them with RTSPS.
